### PR TITLE
Week 2 Submission by Yan:

### PR DESCRIPTION
**Pull request by YanCotta**
Preprocessing scripts and README with instructions / steps I took

### CODE REVIEW

**Comment by YanCotta** [comment]
This pull request includes my Week 2 submission for data preprocessing. The `submissions-team/Yan/` folder contains:
- Scripts for dataset extraction, image verification, class analysis, and train/validation splitting.
- A class distribution plot (`class_distribution.png`).
- A README with instructions to reproduce the work.
The dataset was not included due to its size, but the README explains how to download and process it.

**Comment by david-boules** [comment]
Nice work, Yan. I like the classes visualization and the neat file handling. I would just recommend that you include an additional 'test' split for the data, so that way we are able to evaluate the final model's performance on completely unseen data, after all the model tuning.

**Comment by YanCotta** [comment]
Hey David! Thank you! About your observation, if you check the end of the
"data_preprocessing.py" script (I've attached a snippet below) , you'll see
that that function has already been added! (splitting into training and
test sets).

# Split dataset
if not os.path.exists("split_dataset"):
splitfolders.ratio("temp_dataset", output="split_dataset", seed=42, ratio=(
0.8, 0.2), group_prefix=None, move=False)

# Load datasets
train_dataset = datasets.ImageFolder("split_dataset/train", transform=
train_transforms)
val_dataset = datasets.ImageFolder("split_dataset/val", transform=
val_transforms)

# Create data loaders
train_loader = DataLoader(train_dataset, batch_size=32, shuffle=True,
num_workers=2)
val_loader = DataLoader(val_dataset, batch_size=32, shuffle=False,
num_workers=2)

print(f"Training samples: {len(train_dataset)}")
print(f"Validation samples: {len(val_dataset)}")


Best,

Yan Cotta

*Artificial Intelligence & Data Science: Agentic AI, LLMs, Deep Learning,
Machine Learning, MLOps & LLMOps | Biological & Cognitive Sciences*
LinkedIn <https://www.linkedin.com/in/yan-cotta/> X <https://x.com/CottaYan>
 Medium ***@***.***> GitHub <https://github.com/YanCotta>

**Comment by david-boules** [comment]
Hey Yan. Thanks for your message. I may not have specified, but I was referring to a **third** split of your dataset, so we would have the following:

1. training set - images used in training
2. validation set - unseen images used to evaluate model performance and tune hyperparameters
3. test set - completely unseen images used _when the model has been finalized_ to evaluate the final performance of the model (i.e. it's used to 'showcase' the model in action)

I've noticed that the terms 'test' and 'validation' are used interchangeably when it comes to discussions like these, so I hope that clears things up as to what I was initially suggesting. I stumbled across this wikipedia article about train/validation/test sets, and some of the confusion between these terms, when I was writing this message. (Here it is if you'd like to check it out: https://en.wikipedia.org/wiki/Training,_validation,_and_test_data_sets)
